tests/suites/test_suite_ssl.data

```diff
@@ -930,35 +930,35 @@ depends_on:MBEDTLS_SSL_PROTO_TLS1_2
 ssl_session_serialize_version_check:0:0:0:1:0:MBEDTLS_SSL_VERSION_TLS1_2
 
 TLS 1.3: CLI: session serialization: Wrong major version
-depends_on:MBEDTLS_SSL_PROTO_TLS1_3:MBEDTLS_SSL_CLI_C
+depends_on:MBEDTLS_SSL_PROTO_TLS1_3:MBEDTLS_SSL_CLI_C:MBEDTLS_SSL_SESSION_TICKETS
 ssl_session_serialize_version_check:1:0:0:0:MBEDTLS_SSL_IS_CLIENT:MBEDTLS_SSL_VERSION_TLS1_3
 
 TLS 1.3: CLI: session serialization: Wrong minor version
-depends_on:MBEDTLS_SSL_PROTO_TLS1_3:MBEDTLS_SSL_CLI_C
+depends_on:MBEDTLS_SSL_PROTO_TLS1_3:MBEDTLS_SSL_CLI_C:MBEDTLS_SSL_SESSION_TICKETS
```

Comment:
Because in ssl_tls.c, if MBEDTLS_SSL_SESSION_TICKETS is not defined, `ssl_tls13_session_save/load()` are just returning `MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE`. The reason is that in TLS 1.3 session resumption is supported only through tickets, not though session id.

Reply:
Ok, I see. Seems like something that the documentation of `mbedtls_ssl_session_save/load()` should mention. I understand the reasoning but I can also see how that can surprise users: say I'm using 1.2 in a config without tickets (using cache-based resumption), I'm calling those functions unconditionally client-side and they work; now I enable 1.3 (or inherit it from the default config), so 1.3 is negotiated and boom `mbedtls_ssl_session_save()` starts returning an error. So, I'd double-check this function's documentation and when finding no hint there I'd probably not be super happy.

Reply:
Thanks for raising this. Looking at this, it seems to me that it would also make sense to return in error when `mbedtls_ssl_get_session()` is called on an TLS 1.3 session and MBEDTLS_SSL_SESSION_TICKETS is disabled (not the case currently). What do you think?

Reply:
I don't know. On one hand, I see your point, and I can't think of a reason why people would want to use `get_session()` with TLS 1.3 without tickets - you can't resume anyway. On the other hand, part of me is worried that I have less imagination than users and they might still want to do something with it.

So, for 3.6 I'm inclined to only document what wasn't working anyway (save/load) but not artificially break what was working (get) even if we think it's not useful anyway.

For 4.0 as discussed previously I'd like to refactor this whole API and combine get+save and load+set into a single function each, so that users don't have to manipulate `ssl_session` directly.

Wdyt?

Reply:
That sounds sensible to me, thanks. I will update the documentation of save/load.